packages/indexer/indexer/db.py

```diff
@@ -0,0 +1,168 @@
+import json
+import aiohttp
+
+import asyncpg
+from indexer.chain_mapper import CosmosChain
+from indexer.data import get_block
+
+
+async def drop_tables(pool: asyncpg.pool):
+    await pool.execute(
+        """
+        DROP TABLE IF EXISTS raw CASCADE;
+        DROP TABLE IF EXISTS blocks CASCADE;
+        DROP TABLE IF EXISTS txs CASCADE;
+        """
+    )
+
+
+async def create_tables(pool: asyncpg.pool):
+    async with pool.acquire() as conn:
+
+        await conn.execute(
+            """
+            CREATE TABLE IF NOT EXISTS raw (
+                chain_id TEXT NOT NULL,
+                height BIGINT NOT NULL,
+                block JSONB,      
+                txs JSONB,
+                          
+                PRIMARY KEY (chain_id, height)
+            );
+        """
+        )
+
+        await conn.execute(
+            """
+              CREATE TABLE IF NOT EXISTS blocks (
+                height BIGINT NOT NULL,
+                chain_id TEXT NOT NULL,
+                time TIMESTAMP NOT NULL,
+                block_hash TEXT NOT NULL,
+                proposer_address TEXT NOT NULL,
+                
+                PRIMARY KEY (chain_id, height)
+            );
+            CREATE TABLE IF NOT EXISTS txs (
+                txhash TEXT NOT NULL PRIMARY KEY,
+                chain_id TEXT NOT NULL,
+                height BIGINT NOT NULL,
+                tx JSONB,
+                tx_response JSONB,
+                tx_response_tx_type TEXT,
+                code TEXT,
+                data TEXT,
+                info TEXT,
+                logs JSONB,
+                events JSONB,
+                raw_log TEXT,
+                gas_used BIGINT,
+                gas_wanted BIGINT,
+                codespace TEXT,
+                timestamp TIMESTAMP,
+                
+                FOREIGN KEY (chain_id, height) REFERENCES blocks (chain_id, height)
+            );
+            CREATE TABLE IF NOT EXISTS messages (
+                key TEXT,
+                value TEXT,
+                txhash TEXT,
+                
+                FOREIGN KEY (txhash) REFERENCES txs (txhash)
+            );
+            """
+        )
+
+        await conn.execute(
+            """ 
+            CREATE OR REPLACE FUNCTION parse_raw() RETURNS TRIGGER AS $$
+            DECLARE
+                tx JSONB;
+                tx_responses JSONB;
+            BEGIN
+                INSERT INTO blocks (height, chain_id, time, block_hash, proposer_address)
+                VALUES (
+                    NEW.height,
+                    NEW.chain_id,
+                    (NEW.block->'block'->'header'->'time')::TEXT::timestamp without time zone,
+                    (NEW.block->'block_id'->>'hash')::TEXT,
+                    (NEW.block->'block'->'header'->>'proposer_address')::TEXT
+                );
+                
+                        
+                FOR tx_responses IN SELECT * FROM jsonb_array_elements(NEW.txs->'tx_responses')
```

Comment:
This syntax is strange to me

Reply:
It was bizarre to me too but think of it like: for each row we need to normalize the data based on the json. 

I can store the select in a variable tho to make it look cleaner